src/test/java/teammates/ui/webapi/UpdateNotificationActionTest.java

```diff
@@ -0,0 +1,132 @@
+    @Override
+    String getActionUri() {
+        return Const.ResourceURIs.NOTIFICATION;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return PUT;
+    }
+
+    @Test
+    @Override
+    protected void testExecute() throws Exception {
+        final String TEST_NOTIFICATION = "notification1";
+        NotificationAttributes testNotificationAttribute = typicalBundle.notifications.get(TEST_NOTIFICATION);
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_ID, testNotificationAttribute.getNotificationId(),
+        };
+        NotificationUpdateRequest req = getTypicalUpdateRequest();
+        NotificationStyle style = req.getStyle();
+        NotificationTargetUser targetUser = req.getTargetUser();
+        String title = req.getTitle();
+        String message = req.getMessage();
+        String invalidTitle = "";
+        String invalidNotificationId = "InvalidNotificationId";
+
+        loginAsAdmin();
+
+        ______TS("Typical Case: Update notification successfully");
+        req = getTypicalUpdateRequest();
+        UpdateNotificationAction action = getAction(req, requestParams);
+        NotificationData res = (NotificationData) action.execute().getOutput();
+
+        NotificationAttributes updatedNotification = logic.getNotification(res.getNotificationId());
+
+        // Verify that correctly updated in the DB
+        assertEquals(req.getStartTimestamp(), updatedNotification.getStartTime().toEpochMilli());
+        assertEquals(req.getEndTimestamp(), updatedNotification.getEndTime().toEpochMilli());
+        assertEquals(style, updatedNotification.getStyle());
+        assertEquals(targetUser, updatedNotification.getTargetUser());
+        assertEquals(title, updatedNotification.getTitle());
+        assertEquals(message, updatedNotification.getMessage());
+
+        ______TS("Parameters cannot be null");
+        req = getTypicalUpdateRequest();
+        req.setStyle(null);
+        InvalidHttpRequestBodyException ex = verifyHttpRequestBodyFailure(req, requestParams);
+        assertEquals("Notification style cannot be null", ex.getMessage());
+
+        req = getTypicalUpdateRequest();
+        req.setTargetUser(null);
+        ex = verifyHttpRequestBodyFailure(req, requestParams);
+        assertEquals("Notification target user cannot be null", ex.getMessage());
+
+        req = getTypicalUpdateRequest();
+        req.setTitle(null);
+        ex = verifyHttpRequestBodyFailure(req, requestParams);
+        assertEquals("Notification title cannot be null", ex.getMessage());
+
+        req = getTypicalUpdateRequest();
+        req.setMessage(null);
+        ex = verifyHttpRequestBodyFailure(req, requestParams);
+        assertEquals("Notification message cannot be null", ex.getMessage());
+
+        ______TS("Timestamps should be greater than 0");
+        req = getTypicalUpdateRequest();
```

Comment:
Same as null params, I think it is safe to assume that the timestamp will be > 0, if not `getAndValidateRequestBody` itself would throw `InvalidHttpRequestBodyException`. However it will be useful to test that start time should be before end time.

Reply:
Can add this test case as well, but same as null params, should the existing test be removed?